aspnetcore/blazor/components/index.md

```diff
@@ -783,6 +783,102 @@ In the preceding example:
 > [!IMPORTANT]
 > If a Razor component defines an event that's triggered from a background thread, the component might be required to capture and restore the execution context (<xref:System.Threading.ExecutionContext>) at the time the handler is registered. For more information, see [Calling `InvokeAsync(StateHasChanged)` causes page to fallback to default culture (dotnet/aspnetcore #28521)](https://github.com/dotnet/aspnetcore/issues/28521).
 
+:::moniker-end
+
+:::moniker range=">= aspnetcore-8.0"
+
+### Dispatch exceptions to Blazor's synchronization context
+
+Use `ComponentBase.DispatchExceptionAsync` in Razor components to process exceptions thrown outside of Blazor's synchronization context. This permits the app's code to treat the exceptions as through they're lifecycle method exceptions. Thereafter, Blazor's error handling mechanisms, such as [error boundaries](xref:blazor/fundamentals/handle-errors#error-boundaries), can process the exceptions.
```

Comment:
It's not about being outside the sync context. It's about being outside the lifecycle methods. People should use
DispatchExceptionAsync when:

 * Some code *outside the call stack of a lifecycle method* causes an exception, which they catch
 * They want it to be handled as if it was within the lifecycle of a specific component instance (for example, triggering the error boundaries for that component)

This is true independently of whether their code is running on the sync context. The sync context doesn't come into it.